examples/plugins/traffic-control/main.go

```diff
@@ -0,0 +1,178 @@
+	"path/filepath"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// TODO:
+//
+// do not try to install the qdics on the network interface every
+// time, skip this step if it is already installed (currently we do
+// "replace" instead of "add", but this check may be a way of avoiding
+// more one-time installation steps in future).
+//
+// somehow inform the user about the current traffic control state
+// (either add some metadata about latency or maybe add background
+// color to buttons to denote whether the button is active); this may
+// involve sending shortcut reports as a part of a response to the
+// control request
+//
+// detect if ip and tc binaries are in $PATH
+//
+// detect if required sch_netem kernel module is loaded; note that in
+// some (rare) cases this might be compiled in the kernel instead of
+// being a separate module; probably check if tc works, if it does not
+// return something like "not implemented".
+//
+// add traffic control on ingress traffic too (ifb kernel module will
+// be required)
+//
+// currently we can control latency, add controls for packet loss and
+// bandwidth
+//
+// port to eBPF?
+
+type containerClient interface {
+	Start()
+}
+
+type Plugin struct {
+	reporter *Reporter
+
+	clients []containerClient
+}
+
+func main() {
+	const socket = "/var/run/scope/plugins/traffic-control.sock"
+
+	setupSignals(socket)
+
+	listener, err := setupSocket(socket)
+	if err != nil {
+		log.Fatalf("Failed to setup socket: %v", err)
+	}
+	defer func() {
+		listener.Close()
+		os.Remove(socket)
+	}()
+
+	plugin, err := NewPlugin()
+	if err != nil {
+		log.Fatalf("Failed to create a plugin: %v", err)
+	}
+	if err := plugin.Serve(listener); err != nil {
+		log.Fatalf("failed to serve: %v", err)
+	}
+}
+
+func setupSignals(socket string) {
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt)
+	go func() {
+		<-interrupt
+		os.Remove(socket)
+		os.Exit(0)
+	}()
+}
+
+func setupSocket(socket string) (net.Listener, error) {
+	os.Remove(socket)
+	if err := os.MkdirAll(filepath.Dir(socket), 0755); err != nil {
+		return nil, fmt.Errorf("failed to create directory %q: %v", filepath.Dir(socket), err)
+	}
+	listener, err := net.Listen("unix", socket)
+	if err != nil {
+		return nil, fmt.Errorf("failed to listen on %q: %v", socket, err)
+	}
+
+	log.Printf("Listening on: unix://%s", socket)
+	return listener, nil
+}
+
+func NewPlugin() (*Plugin, error) {
+	store := NewStore()
+	dockerClient, err := NewDockerClient(store)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a docker client: %v", err)
+	}
+	reporter := NewReporter(store)
+	plugin := &Plugin{
+		reporter: reporter,
+		clients: []containerClient{
+			dockerClient,
+		},
+	}
+	for _, client := range plugin.clients {
+		go client.Start()
+	}
+	return plugin, nil
+}
+
+func (p *Plugin) Serve(listener net.Listener) error {
+	http.HandleFunc("/report", p.report)
+	http.HandleFunc("/control", p.control)
+	return http.Serve(listener, nil)
+}
```

Comment:
Probably nicer to instantiate a ServeMux, and use that to serve here, instead of modifying the global http handler.